llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -2763,6 +2763,41 @@ static Instruction *foldSelectWithFCmpToFabs(SelectInst &SI,
     }
   }
 
+  // Match select with (icmp slt (bitcast X to int), 0)
+  //                or (icmp sgt (bitcast X to int), -1)
```

Comment:
I suppose it would also work for `fptosi`?

Reply:
Yeah. But this pattern is usually generated by `std::signbit`. Not sure that `((fptosi X) <s 0 ? -X : X)` exists in real-world applications.

Reply:
I agree the bitcast one is probably alot more likely, but could imagine some template code casting float -> int for a compare.